SPHINXsys/src/for_3D_build/geometries/complex_shape_image_mesh.h

```diff
@@ -0,0 +1,90 @@
+/* -------------------------------------------------------------------------*
+*								SPHinXsys									*
+* --------------------------------------------------------------------------*
+* SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle	*
+* Hydrodynamics for industrial compleX systems. It provides C++ APIs for	*
+* physical accurate simulation and aims to model coupled industrial dynamic *
+* systems including fluid, solid, multi-body dynamics and beyond with SPH	*
+* (smoothed particle hydrodynamics), a meshless computational method using	*
+* particle discretization.													*
+*																			*
+* SPHinXsys is partially funded by German Research Foundation				*
+* (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1				*
+* and HU1527/12-1.															*
+*                                                                           *
+* Portions copyright (c) 2017-2020 Technical University of Munich and		*
+* the authors' affiliations.												*
+*                                                                           *
+* Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+* not use this file except in compliance with the License. You may obtain a *
+* copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+*                                                                           *
+* --------------------------------------------------------------------------*/
+/**
+* @file complex_shape_image_mesh.h
+* @brief x 
+* @details x 
+*			x 
+* @author	Yijin Mao
+*/
+
+#ifndef COMPLEX_SHAPE_IMAGE_MESH_H
+#define COMPLEX_SHAPE_IMAGE_MESH_H
+
+#define _SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING
+
+#include "base_geometry.h"
+#include "SimTKcommon.h"
+#include "SimTKmath.h"
+#include "Simbody.h"
+#include "simbody_middle.h"
+#include "geometry.h"
+#include "complex_shape_mesh.h"
+//#include "image_mesh_shape.h"
+
+#include <iostream>
+#include <string>
+#include <fstream>
+
+/** Macro for APPLE compilers*/
+#ifdef __APPLE__
+#include <boost/filesystem.hpp>
+namespace fs = boost::filesystem;
+#else
+#include <experimental/filesystem>
```

Comment:
when should we switch to std::filesystem in c++17? @Xiangyu-Hu 
Clang is complaining this when we do webassembly compilation